Implement checking for nested alias content & Clean-up validators

jiuka opened the pull request:
### Link to API

https://docs.opnsense.org/development/api/core/firewall.html

### Description

Content validation was disabled for `host` and `network` aliases to allow using nested values. This patch uses the already fetched list of aliases to validate nested entries for `post`, `host` and `network`.

Further the host validation has been improved and some tests added.

jiuka committed:
Implement checking for nested alias content

jiuka committed:
Simplify and improve validation

jiuka committed:
Rework validation to align with opnsense

jiuka committed:
Fix nested alias validation

jiuka committed:
Fix yamllint warning

ansibleguy commented:
I hope I find time this weekend to review this. Sorry for the long delay..

superstes committed:
chore: migrate validators

superstes commented:
@jiuka As I've recognized that we have duplicated validator-functions - I've migrated those to be inside the same `validators.py` file.

We still have some duplicate logic in the port-checks. Maybe you could give it a look?

superstes commented:
The port-validation is also an issue in https://github.com/O-X-L/ansible_opnsense/issues/183

jiuka committed:
Fix tests after validators migration

jiuka committed:
Split port and port range validation

Most port values expect one port number. Allowing a range or any is the exception.

jiuka committed:
Merge branch 'latest' into nested_aliases

jiuka committed:
Fix linterwarnings

jiuka committed:
migrate validators for new modules

superstes committed:
lint: disable false-positive

superstes committed:
chore: remove old templates

superstes committed:
chore: move field-validations to base-class, replace simple port-validation by int-field-validation

superstes committed:
chore: remove non-working test-log-links

superstes committed:
test: fix tests

superstes commented:
The tests (except openvpn) did pass.
@jiuka Could you review the changes?
If there are no issues left I would like to merge this soon

jiuka commented on docs/source/usage/4_develop.rst:
The [line 365](https://github.com/scsitteam/collection_opnsense/blob/68b081247417a7c1405b5709b7239ed2a38ca9f8/docs/source/usage/4_develop.rst?plain=1#L365) should be removed/updated too.

jiuka commented on plugins/module_utils/main/unbound_forward.py:
```diff
@@ -33,7 +33,8 @@ class Forward(BaseModule):
     STR_LEN_VALIDATIONS = {
-        'description': {'min': 0, 'max': 255}
+        'description': {'min': 0, 'max': 255},
+        'port': {'min': 1, 'max': 65535},
```
This should probably be a `INT_VALIDATIONS` not a `STR_LEN_VALIDATIONS`

jiuka reviewed (commented):
Just two small things. :+1:

superstes committed:
fix: unbound-fwd validation + remove docs tmpl-reference

ansibleguy committed:
Merge branch 'latest' into nested_aliases

superstes commented:
Thank you for catching those :+1:

superstes committed:
test: lint fixes
